src/commands/index.rs

```diff
@@ -0,0 +1,163 @@
+use serde_json::Value;
+
+use redismodule::{Context, RedisError, RedisResult, RedisValue};
+use redismodule::{NextArg, REDIS_OK};
+
+use redisearch_api::{Document, FieldType};
+
+use crate::error::Error;
+use crate::redisjson::{Format, RedisJSON};
+use crate::schema::Schema;
+use crate::REDIS_JSON_TYPE;
+
+pub mod schema_map {
+    use crate::schema::Schema;
+    use std::collections::HashMap;
+
+    type SchemaMap = HashMap<String, Schema>;
+
+    /// We keep a static map, since it needs to be accessed from multiple Redis module commands
+    /// and there is no other obvious way to implement this (such as a "user data" pointer maintained
+    /// for us by Redis).
+    ///
+    /// The init function should be called only once.
+    /// Unwrapping the Option is thus safe afterwards.
+    ///
+    /// Since we have only one thread at the moment, getting a (mutable) reference to the map
+    /// is also safe. If we add threads, we can simply wrap the map: `Option<Mutex<SchemaMap>`.
+    ///
+    static mut SCHEMA_MAP: Option<SchemaMap> = None;
+
+    pub fn init() {
+        let map = HashMap::new();
+        unsafe {
+            SCHEMA_MAP = Some(map);
+        }
+    }
+
+    pub fn as_ref() -> &'static SchemaMap {
+        unsafe { SCHEMA_MAP.as_ref() }.unwrap()
+    }
+
+    pub fn as_mut() -> &'static mut SchemaMap {
+        unsafe { SCHEMA_MAP.as_mut() }.unwrap()
+    }
+}
+
+///////////////////////////
+
+fn add_field<I>(index_name: &str, field_name: &str, mut args: I) -> RedisResult
+where
+    I: Iterator<Item = String>,
+{
+    let path = args.next_string()?;
```

Comment:
@gavrie why not just pass the path as argument to the function?

Reply:
Agreed. I originally wanted the function signatures to be symmetric and handle `args` in a standard way, but that's overly complicated and it would indeed be simpler to pass the `path` directly.